Breathing Pluox while sleeping slowly heals organ damage

GuillaumePrata opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Alternative to #67990 and as that one was merged, it would be a revert.

Breathing Pluoxium now heals organs slowly if the target is sleeping instead of healing suffocation damage at all times.

I also did some code clean up while there and moved the atmos reagents to their own file because `other_reagents` had almost 3000 lines...
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
I really dislike that Pluoxium has been mostly an O2 but better, it makes it an useful but really boring gas...

I didn't want to buff gamer internals so adding an effect to Pluoxium would be hard.
Lucky for me Cryotubes got reworked with support for atmos canisters so it is a good place to give Pluoxium a utility and make a new way for other departments to interact with atmos that isn't "Yay, breach/plasmafire is killing me again!!"
It also makes it be an entry level for Atmos-Medbay cooperation while Helium is an advanced option.

If doctors get atmostechs to cooperate this will cut on the brain draining side of Medbay's job when a serious emergency pops up and multiple people are dead, doing organ surgery/swap again, and again, and again...
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Guillaume Prata
balance: Breathing Pluoxium will inject its chemical reagent that slowly heals organ damage if the target is sleeping. Hint: Cryotubes.
balance: Breathing Pluoxium won't heal suffocation damage anymore.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Kapu1178 commented:
us this really worth the performance cost? Life() is already agonizing. 

Also now would be a great time to do generalized gas > reagent inhalation :)))))))

GuillaumePrata commented:
> Will this heal organs that are nonfunctional due to organ damage?

If you can manage to keep someone with failing organs alive long enough to put them on cryo and keep them alive to process it, yes.

moocowswag commented:
Pluox internals + thermal vision!!!

Colovorat commented:
> Pluox internals + thermal vision gene!!!

You first make video on it, then you report a bug. 
Do i really have to teach you such basics?

private-tristan commented:
if medical is at the point where organ damage is a problem, the first organ med docs will need to replace is the heart (to defib) and while the organ manip surgery is going on they can replace lungs liver etc, leaving only brain ears and eyes, which all have their own chems that heal them

I don't really see a point for healing organ damage until organ damage has noticeable downsides (which would be medical hell) other than getting some ominous messages in your char bar 

maybe having it double the healing of things like cryo and healium if you're in cryo tubes?

GuillaumePrata commented:
> and while the organ manip surgery is going on they can replace lungs liver etc

Assuming they have the organs at hand, and still wasting their time/resources if they do.

> I don't really see a point for healing organ damage until organ damage has noticeable downsides (which would be medical hell) other than getting some ominous messages in your char bar

The annoying coughing and other minor side effects of damaged organs is something that Pluox cryotubes would fix and I assume most doctors would like a way to fix these minor issues without having to do a complete new organ transplant for everyone that was dead for a small time.

Mothblocks commented:
No maintainer has expressed interest in this

tommysalami3 commented:
![image](https://user-images.githubusercontent.com/77305289/179384262-ba09408e-f53f-4dff-a037-0eac713a43b8.png)
me watching a perfectly reasonable guill PR get closed because atmos boring

GoldenAlpharex committed:
Fixes the capitalization of the long-range gas analyzer (#69069)

tgstation-server committed:
Automatic changelog generation for PR #69069 [ci skip]

axietheaxolotl committed:
new dead body mapping helper icons (#68915)

new dead body spawner icons

tgstation-server committed:
Automatic changelog generation for PR #68915 [ci skip]

Jacquerel committed:
Splits up 'Gross' food from 'Gore' and reshuffles taste preferences accordingly (#68899)

* Splits up 'gross' food from 'carrion'

* Adds a couple of missed burgers.
Moths dislike Carrion rather than it being poison to them.

* BEES is now BUGS

* It's always safe and a good idea to use find/replace

tgstation-server committed:
Automatic changelog generation for PR #68899 [ci skip]

Wallemations committed:
Changes sprites for the two bat sub-variants (#68882)

Updates the sprite for the metal & homerun bat

tgstation-server committed:
Automatic changelog generation for PR #68882 [ci skip]

ReinaCoder committed:
moves the centcom jacket's mob sprite to jacket.dmi instead of spacesuit.dmi (#68984)

Moves the centcom jacket's mob sprite to jacket.dmi instead of spacesuit.dmi

ShizCalev committed:
Adds QM to authority & conspiracy phobias, QM's bedsheets will now make you dream of authority and SILVERY IDs (#69037)

Adds QM to authority & conspiracy phobias

tgstation-server committed:
Automatic changelog generation for PR #69037 [ci skip]

NanoTrasen-Inc committed:
Adds 2 new sounds to the Synthesizer's 'Fun' category (#68978)

Adds 2 new sounds to the 'Fun' category of the synthesizer

MeowSynth
Spaceman

tgstation-server committed:
Automatic changelog generation for PR #68978 [ci skip]

XyzzyThePretender committed:
adjusts metabolism for munchies (#69009)

* adjusts metabolism for munchies (weed metabolism). Now it is similar to running non-stop.

tgstation-server committed:
Automatic changelog generation for PR #69009 [ci skip]

VioletN committed:
Buffs the chilling effects of void heretic powers, fixes void storm visuals not updating (#68896)

* Buffs the chilling effects of void heretic powers

- The void Mansus Grasp and Mark of Void heretic abilities now inflict a status condition called void chill, which steadily lowers the victim's body temperature and also inflicts a mild slowdown.
- The void storm summoned by ascended void heretics no longer directly inflicts burn damage, but now lowers body temperature much more strongly.
- The void storm visual overlays now properly update when the heretic exits an area, making them a more accurate indicator of which area the heretic is in.

* Split heretic status effects into their own files, code improvements

* split the buffs as well

* Adds a bit of burn damage back to void storm

tgstation-server committed:
Automatic changelog generation for PR #68896 [ci skip]

Zonespace27 committed:
Butcher's meathook is no longer restricted by TRAIT_NOGUNS (#68989)

can_trigger_gun always returns TRUE on meathooks, because they aren't real guns so shouldn't be checking if guns can be used.

tgstation-server committed:
Automatic changelog generation for PR #68989 [ci skip]

ShizCalev committed:
Fixes kidnap traitor objective failing to generate if the tatertot failed a previous kidnapping objective (#69051)

tgstation-server committed:
Automatic changelog generation for PR #69051 [ci skip]

ShizCalev committed:
Fixes incise heart surgery runtime (#69032)

* Fixes incise heart surgery runtime

tgstation-server committed:
Automatic changelog generation for PR #69032 [ci skip]

XyzzyThePretender committed:
Newly built turrets start unlocked (#69012)

* newly build turrets are unlocked + code style

tgstation-server committed:
Automatic changelog generation for PR #69012 [ci skip]

JohnFulpWillard committed:
Fixes calomel's description to state toxic chems (#69026)

Calomel was changed to only remove toxic chemicals in https://github.com/tgstation/tgstation/commit/51867bb116deb4a58203a8f91f7e24813b7a1952 yet their description wasn't updated to match this.

tgstation-server committed:
Automatic changelog generation for PR #69026 [ci skip]

MrMelbert committed:
Adds support for "realistic" public elevators (elevator doors, elevator panel, etc) (#68888)

* Elevators are a bit more friendly part 1
- Emaggable elevator buttons / support for emaggable device assemblies
- Delayed travel between floors
- Elevators can now show a warning sign to areas below while travelling
- Elevators can now optionally wound heavily instead of gibbing

* Adds supoprt for "lift doors"

* Comment tweak

* Splitting these variables

* Functional prototype

* multiz debug

* Elevator button framework

* Unifies these behaviors

* Emergency doors

* Lift button framework

* UI closes on floor change

* Testing changes

* Fix

* UI Tweaks

* Panel works pretty swell

* Minor tweaks

* Move to static

* Bonus tram change

* User experience

* Slight tweak to mapload stuff

* This is silly

* Some UI tweaks, need to update css

* CSS and ui overhaul

* Documentation updates.

* Multi-z lift support

* Multitile lift fix

tgstation-server committed:
Automatic changelog generation for PR #68888 [ci skip]

orthography committed:
Emitter status shows minimum and maximum firing delay (#68921)

About The Pull Request

Emitter status display on examine will now display the minimum and maximum firing delay depending in place of an inaccurate static speed.

Why It's Good For The Game

Fixes #68820

As the emitter shoots at a random speed (inexplicably, it seems to create more issues than it solves), saying it'll fire "one beam every x seconds" is inaccurate as it's subject to change.

tgstation-server committed:
Automatic changelog generation for PR #68921 [ci skip]

XyzzyThePretender committed:
Fix grammar for solar panel construction (#69010)

About The Pull Request

Fixes #69001

"into place" only shows up when securing a solar panel using ? : conditional.
Why It's Good For The Game

Grammar fix.

Co-authored-by: san7890 <the@san7890.com>
Co-authored-by: tattle <66640614+dragomagol@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69010 [ci skip]

Mooshimi committed:
[GBP No Update] Perish, individual logging runtime (#69024)

missed 2 or 3(lol it was more when I look back at the files), LOG_GAME tags on the log_message line, and did some cleaning up since i was looking through every log_message again

Co-authored-by: tattle <66640614+dragomagol@users.noreply.github.com>

Co-authored-by: tattle <66640614+dragomagol@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69024 [ci skip]

Time-Green committed:
The Great Species Dedatumming 2: Makes external organs operable (#68877)

Finally makes external organs (horns, moth wings, antennae etc.) operable!
Also adds a new surgery: feature manipulation. It's a shorter version of organ manipulation that only lets you operate on species features. Note that organ manipulation can operate only on internal organs.

tgstation-server committed:
Automatic changelog generation for PR #68877 [ci skip]

dragomagol committed:
Codeowners housekeeping (#69124)

codeowners housekeeping

Co-authored-by: tattle <article.disaster@gmail.com>

MrMelbert committed:
Blackbox logs heretic starting knowledge and ascensions (#69099)

tgstation-server committed:
Automatic changelog generation for PR #69099 [ci skip]

Time-Green committed:
fixes golden wheelchair never spawning (#69094)

* The Golden wheelchair will now spawn if you have over 5000 hardcore random score, as it was supposed to.

tgstation-server committed:
Automatic changelog generation for PR #69094 [ci skip]

Timberpoes committed:
Spends 6 hours teaching Papercode how to conform to better standards and practices, and also fixes a trivial 5 minute bug with command reports. (#69071)

tgstation-server committed:
Automatic changelog generation for PR #69071 [ci skip]

ShizCalev committed:
fixes runtime when clicking an airlock with TK (#69056)

* fixes runtime when clicking an airlock with TK by checking the accessor is a mob.

actions-user committed:
Automatic changelog compile [ci skip]

vincentiusvin committed:
[NO GBP] Fixed an edge case when SM delam messages failed to update. (#68968)

About The Pull Request

Forgot to update the last delam var.

tgstation-server committed:
Automatic changelog generation for PR #68968 [ci skip]

ShizCalev committed:
makes roulette tables NFC enabled (#69100)

* Roulette Table - PDAs and cards can now play roulette too.

tgstation-server committed:
Automatic changelog generation for PR #69100 [ci skip]

GoldenAlpharex committed:
Fixes conveyor belts not updating their sprites immediately when modified with a screwdriver (#69068)

Fixes conveyor belts not updating their sprites immediately when interacted with with a screwdriver

tgstation-server committed:
Automatic changelog generation for PR #69068 [ci skip]

GoldenAlpharex committed:
Fixes an oversight in food exports (#69106)

tgstation-server committed:
Automatic changelog generation for PR #69106 [ci skip]

vincentiusvin committed:
Simplifies the SM transmission calculation a bit. (#68972)

Reframes the existing interactions into the transmission modifier framework. The code there is more robust than snowflaked separate checks like this.

BZ: #45676 Looks like it was intended to be a quite good gas for power buffing, but it has lost its intended purpose over the years. Current status quo is that it strengthens the modifier, either positive or negative. I've decided to use a pure BZ sm as a baseline, so its fully negative now.

Water: #49624 Intended to nerf the SM power, but it ended up as a transmission dampener (opposite of BZ) because of changes to the SM over the years. So it makes the weak a bit less weak and the strong a bit less strong. A pure water vapor has zero effect on power (since transmit modifier is zero) so i cant use it as a baseline. So i followed the author's original intention of it being a bad for power gas.

Freon: Nukes power when above 30%. Changed it to 33% because the transmit numbers are a bit nicer. It checks out try it in your head. (1 + (1/3 * -30 / 10) = 0. Difference is its affected by other gases now

CO2: Same with freon, barely any changes. Apparently doubles power when at 100% comp. Affected by other gases now.

Easier to maintain and explain, and we don't really lose out on much. I attempted to preserve the behavior as best as i can.

To be perfectly fair I have no quarrels with the number at all, don't really care which one is stronger than which. Just want to keep the old behavior. If you want any of them changed LMK.

tgstation-server committed:
Automatic changelog generation for PR #68972 [ci skip]

Pepsilawn committed:
Moves sinks' layer to above the object layer (#68964)

This PR moves sinks to layer 3.2 as opposed to 3, the object layer.

Sinks are coded in with a pixel displacement and kitchen sinks are large enough for the collision check to fail if the sink is against a solid structure on an above layer, such as a window or an airlock.


Another big issue is that it looks bad

This is at the cost of 3 pixels appearing on top of a window if placed facing south against one, so this isn't a complete fix to the aesthetic of it, the primary focus of the PR is to make the sink interactable, and even then I believe it's a better trade-off:

tgstation-server committed:
Automatic changelog generation for PR #68964 [ci skip]

MrMelbert committed:
Fixes 118(give or take) cases of mapload not being passed to initilaize (#69107)

fixes 114 cases of mapload not being passed to initilaize

SuperNovaa41 committed:
Replaces the mood component with a mood datum (#68592)


About The Pull Request

Mood was abusing signals and get component pretty badly, so I redid it as a datum to stop this.
Why It's Good For The CODEBASE

Better code pratices, also gives admins easier tools to manage mood
Changelog

cl
admin: Added two new procs into the VV dropdown menu to add and remove mood events from living mobs.
/cl

tgstation-server committed:
Automatic changelog generation for PR #68592 [ci skip]

necromanceranne committed:
Nukies Update 5: Bulldog Unchained (Wardens hate this) (I gave it a second magazine) (#69119)

Gives the Bulldog Shotgun a secondary magazine. You can press rightclick to fire and swap to the secondary magazine. You can also swap the magazine manually by pressing Alt-Rightclick on the gun, and load in a new magazine by rightclicking the gun with the magazine.

tgstation-server committed:
Automatic changelog generation for PR #69119 [ci skip]

AnturK committed:
Fixes some fontawesome weirdness. (#69120)

Fixes #69087
FA dropped support for IE so they used var() defaults for most constants which need replacing.

JohnFulpWillard committed:
Removes ComponentInitialize() (#69118)

* Removes ComponentInitialize()

Completely removes ComponentInitialize() as a proc, which was called on every single atom in the game, twice in some instances (like new players), over something that can already be done with Initialize().
This is the second attempt at doing this, after the first attempt fell apart for some reason. This time it was way easier though, since storages are no longer a Component.

* update icon blocker added before calling parent

* Update code/game/machinery/porta_turret/portable_turret.dm

Co-authored-by: san7890 <the@san7890.com>

* adds a mapload while I'm here

* moves human mood

* Does some UNRELATED thing to the PR

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Co-authored-by: san7890 <the@san7890.com>
Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

JohnFulpWillard committed:
Species has custom brute/burn/cloning damage examine (#69029)

About The Pull Request

Different human species can now have different examine texts, allowing androids to not have bruising, and instead have denting, like a cyborg would.

Why It's Good For The Game

It's a little more customization to some species who don't exactly have flesh and bone to be bruised and burned normally.

tgstation-server committed:
Automatic changelog generation for PR #69029 [ci skip]

Ryll-Ryll committed:
Hotfixes outdated mood datum lines (#69149)

I was an idiot when I merged #69118 without double checking the build tasks, as it conflicted with changes made in #68592 that removed the mood datums. That's 100% on me.

This PR just removed the lines that were meant to be removed/replaced
Why It's Good For The Game

We can compile again

Spookuni committed:
Fixes the privacy bolts on two of the Tram station dorms rooms (#69116)

tgstation-server committed:
Automatic changelog generation for PR #69116 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

MrMelbert committed:
Fixes a copy+paste error with ladder screentips (#69108)

About The Pull Request

/obj/structure/ladder/unbreakable/rune had a copy+past error, so either all ladders were showing as "warp up / down" or all ladder runes were showing as "climb up / down", I'm not sure which as I don't use screentips, I just saw the dupe definition lint while doing the mapload PR

tgstation-server committed:
Automatic changelog generation for PR #69108 [ci skip]

SuperNovaa41 committed:
Adds myself to code owners (#69162)

I reworked mood and forensics so i figured i'd add myself to the
codeowners file :)

Y0SH1M4S73R committed:
Lua UI improvements (#68887)

Atomized from #68816, with a little addition. Fixes some dumb formatting issues with the lua editor, adds a "jump to bottom" button when viewing the state log, and paginates the state logs.

tgstation-server committed:
Automatic changelog generation for PR #68887 [ci skip]

JohnFulpWillard committed:
removes InitializeAIController(), moves it to Initialize (#69129)

The description of InitializeAIController() states that it is meant to be overwritten to assign variables before setting their AI, but this is never actually needed, so this is useless proc overhead, especially when this is called on the Init of every single atom in the game.

567Turtle committed:
mime emote consistency update (#69127)

So, you're a mime main and you've just contracted a virus which makes you sneeze, wouldn't be a problem but gasp!, some TG coder 8,000 years ago only made it so you laugh and scream silently, not the other audible emotes, this PR aims to fix that.

tgstation-server committed:
Automatic changelog generation for PR #69127 [ci skip]

strandsofivy committed:
Fix certain orders not working on departmental supply consoles (#69122)

regression or somethign

ShizCalev committed:
Fixes runtime when examining xenos (#69055)

* Fixes runtime when examining xenos

* john

Wallemations committed:
GAGS most radio encryption keys (#69104)

* the radio star

* I think it's funny that I've had the same contrast problem on all of my GAGS PRS

* oops forgot to make the RD's key unique

* fine

* what if I said please

* swaps translation key to be in line with the other new ones

tgstation-server committed:
Automatic changelog generation for PR #69104 [ci skip]

ReinaCoder committed:
Profane Scholar Chaplain outfit resprite (#69103)

profane scholar resprite

tgstation-server committed:
Automatic changelog generation for PR #69103 [ci skip]

Imaginos16 committed:
New Holopad Sprite! (#69126)

* le sprites

* lmao nice

tgstation-server committed:
Automatic changelog generation for PR #69126 [ci skip]

vincentiusvin committed:
Refactors operating tables to be event driven + QoL + Unit Test (#69015)

* Event driven table

* Operating computer fix + loosening of check

* Unit testing

* IT NEEDS TO BE FORCE MOVE YOU GOTTA CLIMB TABLES AAAAH

* Migrated patient to carbon instead of human
Has no real bearing on the experiments tbh

* DNAs can be null apparently

* Simplify replacement code

* Move comments

tgstation-server committed:
Automatic changelog generation for PR #69015 [ci skip]

twilightwanderer committed:
Fax Machine (#69083)

* Adds the Fax machine, which allows you to send and receive faxes from around the station.

Co-authored-by: twilightwanderer <twilightwanderer@users.noreply.github.com>
Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69083 [ci skip]

Mooshimi committed:
blind alert too big for stripping and feeding (#69114)

About The Pull Request

span_userdanger -> span_danger

Why It's Good For The Game

I think it's hilarious that it's that big, but it's not necessary. It only makes sense for blind hit alerts which use span_userdanger for nonblind alerts too.

tgstation-server committed:
Automatic changelog generation for PR #69114 [ci skip]

MrMelbert committed:
Fixes some improper global signal use (payment component, traitor objectives) (#69131)

* Fixes some improper signal use of SEND_GLOBAL_SIGNAL()

tgstation-server committed:
Automatic changelog generation for PR #69131 [ci skip]

Hatterhat committed:
Fixes large crates (e.g. canisters) not abiding by delivery requirements (#69062)

* Large crates that arrive in wooden boxes now properly follow departmental delivery requirements.

Co-authored-by: Hatterhat <Hatterhat@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69062 [ci skip]

JohnFulpWillard committed:
[MDB IGNORE] Refactors drinks and fixes a lot of food problems (#69081)

* Makes condiments their own subtype, fixes geese, prepares for merging

* Fixes geese checking drink type instead of edible foodtype to eat gross food.
* Renames foodtype var on drinks to drink_types to prevent above from happening again because it KEEPS HAPPENING. DRINKS AREN'T FOOD!
* Makes Condiments their own subtype of reagent_containers because they don't make any use of being a subtype of food, at all.
* Starts moving things from food to /food/drink subtype in preparation for merging /food/drink with /drink

* fully removes Food subtype

* /reagent_containers/drinks are now /reagent_containers/cup - This is so it's no longer confused with eachother.
* /food/drinks is now /reagent_containers/cup/drinks, so we can keep their special abilities.
* Fixes a LOT of errors with food, which are STILL checking the reagent_containers, despite ACTUAL food being refactored away from it a long time ago.

This doesn't compile yet, but I do want to make sure my progress is well tracked.

* remove copypaste code, changes soda cans

* Removes most copy paste code between the two drinks, moving most stuff to parent whenever needed.
* Made soda cans their own subtype since they didn't share anything with glass bottles anyways.
* Fixes more problems with food/drinks, especially with geese. Geese really were just broken this whole time and no one said a word...
* Removes a snowflake signal, now that both drink types share a common one.
* Adds everything to the .dme

Currently my goal is to get this all compiling, then remove isGlass var by making glass be all glass ones only.

* Moves all icons into a single drinks dmi

I'm not that great at icon stuff, hopefully I didn't forget/break anything.

* Turns juices into their own subtype

This allows us to let them check for type in molotov, to both get rid of a use of isGlass, and so non-glass non-cartons don't show up as 'carton'.

* fixes compile issues, adds updatepaths

* a better updatepaths

* updates the damn maps now

* properly names the updatepath

* how did that get there

* i suck at handling merge conflicts

* how am i this bad

* code improvement and soda fix

* more fixes

* Don't be a timer

Ports from old food bottles to trans the reagents, rather than add a timer to.

* Merge conflicts and fixes bottle smashing

* Bottle smashing is now consistently functional regardless of how much liquid they have in them, when before it would spill first, then smash on the second hit.

* runs updatepaths again

tgstation-server committed:
Automatic changelog generation for PR #69081 [ci skip]

Imaginos16 committed:
Improved Medical Tools and Medkits by Onule! (#69109)

imageadd: Completely resprites all medkits!
imageadd: Completely resprites all medical tools!

tgstation-server committed:
Automatic changelog generation for PR #69109 [ci skip]

Imaginos16 committed:
New Table Sprites! (#69147)

imageadd: Metal, Reinforced, Reinforced Glass, Plasmaglass, Reinforced Plasmaglass, Titanium Glass, Plastitanium Glass, Wood and Poker tables have been updated!

tgstation-server committed:
Automatic changelog generation for PR #69147 [ci skip]

Jolly-66 committed:
[MetaStation] Adds some missing firelocks between Med and Sci (#69139)

tgstation-server committed:
Automatic changelog generation for PR #69139 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Timberpoes committed:
Fix tape recorder printing out transcripts phat enough to crash your game. (#69121)

About The Pull Request

Tape recorders can hold 10 minutes of say logs and then transcribe them.

People say a lot of things in 10 minutes.

These transcriptions can be quite large.

These transcriptions have a lot of spans, which get sanitised away to nothingness JS-side. This causes a lot of client-side JS processing on paper for transcripts that go super saiyan beyond the character limit.

This can hang peoples' games up when they're running SS13 on their Samsung SmartFridge.

This PR paginates printed transcripts.

tgstation-server committed:
Automatic changelog generation for PR #69121 [ci skip]

ShizCalev committed:
Adds a second tip warning botanists about the dangers of combining the chemical traits of bananas and watermelons :) (#69155)

* Adds another tip warning botanists about the dangers of combining the chemical traits of bananas and watermelons :)

* extra space

ShizCalev committed:
Fixes dental implant pills not working (#69154)

tgstation-server committed:
Automatic changelog generation for PR #69154 [ci skip]

JohnFulpWillard committed:
Fixes some issues with deafness & earhealing (#69136)

1- Replaces checks for deafness as a clothing trait, since clothing traits' have a ref to themselves in the source
2- Makes ear healing not NOT heal if you're above 0 ear damage, because that's the point of fixing your ears?

tgstation-server committed:
Automatic changelog generation for PR #69136 [ci skip]

timothymtorres committed:
Remove deprecated `pocket_storage_component_path` var from clothing (#69166)

Remove deprecated var from clothing

Time-Green committed:
Adds myself to plumbing and external organs codeowners (#69165)

* Adds Time-Green to plumbing and external organs codeowners

* Update .github/CODEOWNERS

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

dragomagol committed:
Makes alerts for blind people being fed/stripped larger (#69171)

Revert "blind alert too big for stripping and feeding (#69114)"

This reverts commit 6b0c6e8dedc95d66a217a3cbb93e9d9e21322f14.

tgstation-server committed:
Automatic changelog generation for PR #69171 [ci skip]

CPTANT committed:
Fixes Modafinil not putting you to sleep despite OD (#69169)

* Fixed Modafinil not putting you to sleep despite OD
https://github.com/tgstation/tgstation/issues/68859

* Now only removes sleep immune when it is time to sleep.

tgstation-server committed:
Automatic changelog generation for PR #69169 [ci skip]

SuperNovaa41 committed:
Adds bear hugging! (#69168)

* Adds bear hugs to the game
adds a way of bear hugging your friends, also adds two new mood events,
a positive and negative bear hug mood event. (Negative only being
applied to those with the bad touch quirk)

* whoops + forgot to save this file

* first person not third

tgstation-server committed:
Automatic changelog generation for PR #69168 [ci skip]

timothymtorres committed:
Refactor hardcoded suit sensor defines (#69164)

* Refactor suit sensor defines

* Add suit sensor define to HoS suit

* Fix sensor_mode define to be accurate

necromanceranne committed:
Fixes bibbles holding normal sized items, cleans up an unused proc (#69163)

tgstation-server committed:
Automatic changelog generation for PR #69163 [ci skip]

SuperNovaa41 committed:
Converts all research categories into defines (#69161)

* Converts all research categories into defines

* missing category + machine categories

* final things i hope

* couple issues i missed

tgstation-server committed:
Automatic changelog generation for PR #69161 [ci skip]

Mothblocks committed:
Vending machines now support categories (and premium/contraband is moved there), adds it to the Clothesmate (#69076)

* Initial setup and data

* Work on UI

* Final UI

* Remove TODO

* Section for categories

tgstation-server committed:
Automatic changelog generation for PR #69076 [ci skip]

GuillaumePrata committed:
Adds "welder crafting" to iron sheets, rods and tiles. (#68987)

* Adds "welder crafting", using welders on iron, floor tiles, and iron rods using LMB/RMB to turn them into eachother. Also adds Balloon alerts while doing so, and contextual tips to indicate this is now a thing.

tgstation-server committed:
Automatic changelog generation for PR #68987 [ci skip]

Imaginos16 committed:
New Sneaker Sprites by Any%! (#69134)

imageadd: Resprites all magboots!
imageadd: Resprites Sneakers!

tgstation-server committed:
Automatic changelog generation for PR #69134 [ci skip]

TheBoondock committed:
Ranged analyzer qol (#68703)

* Ranged analyzers can now read up to 15 tiles away, and will scan the floor you click on, even if you don't directly click on the floor.

tgstation-server committed:
Automatic changelog generation for PR #68703 [ci skip]

robotduinomm committed:
Adds keyboard shell (#69105)

Adds keyboard shell.

Works very similar to the thought listener component for BCIs
It opens up an input panel when you use it (tgui_input_text)
Players with the illiterate quirk cannot use this shell. When they try to use it they get a warning message.
It's printed out as assembled (like compact remote or controller)
It can fit in a bag but not in a box
Has small capacity (SHELL_CAPACITY_SMALL)
Available after researching the Advanced Shells tech node
Can be printed from the component printer or the protolathe
Costs 2000 glass and 10000 iron sheets to print (slightly more than the controller shell)
Has 3 outputs. First one returns the entity who used the shell, second is the text that entity wrote and submitted, third one is triggered after the input window is closed. (by submitting, cancelling or just hitting the X button)
Icons (screenhots from the dream maker)

tgstation-server committed:
Automatic changelog generation for PR #69105 [ci skip]

ShizCalev committed:
Fixes Synthesizer circuit runtime (#69157)

Fixes Synthesizer runtime

spessmanturtle committed:
Add distinguishing features to HFR box icons (#69057)

CapybaraExtravagante committed:
Ambience no longer uses reverb (#69003)

* no more local sound

* setup volume settings

* Update code/controllers/subsystem/ambience.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

* Update ambience.dm

Co-authored-by: Capybara <Capybara@CapybaraMailingServices.com>
Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69003 [ci skip]

Time-Green committed:
fixes error food spawning (#69095)

* fixes error food spawning

* right

* commmmuuuunts

* why am i so fucking stupid

tgstation-server committed:
Automatic changelog generation for PR #69095 [ci skip]

Jolly-66 committed:
[Space Ruins] Fixes the stairs used in the Sulaco Ruin (#69143)

these stairs were silly

tgstation-server committed:
Automatic changelog generation for PR #69143 [ci skip]

timothymtorres committed:
Fix pirates to not be displayed on crew monitors (#69159)

Fix pirates being displayed on crew monitors

tgstation-server committed:
Automatic changelog generation for PR #69159 [ci skip]

Twaticus committed:
Suit DMI split p2: Obj icons (#69174)

About The Pull Request

second part of #68417, which splits the suits.dmi, but this time focuses on obj items. everything stated in that pr applies to here aswell, it follows the same categorization.

Why It's Good For The Game

finishes the much needed splitting of suits.dmi, easing the minds of spriters everywhere and lowering the probability of future pr conflicts in this area

actions-user committed:
Automatic changelog compile [ci skip]

RandomGamer123 committed:
Player-craftable Trapdoors (#68920)

Makes trapdoors player-craftable.
Player-crafted trapdoors have a tiny outline as well as being visible on examine. (Mapped in trapdoors have conspicuous = FALSE by default)
Trapdoors can be made by using a trapdoor kit (crafted via player crafting menu) on openspace.
Trapdoor electronics / assembly can be made via autolathe or engineering lathe.
A preloaded trapdoor remote (optional) can be made by crafting the electronics with a compact remote (from science lathe / circuit lab) and some cable coil.
Also allows unlinking trapdoors via a multitool, decreased trapdoor link range, and made some changes to trapdoor code.

Added change: Trapdoors now don't break with just a crowbar
Why It's Good For The Game

Trapdoors currently can only be added in by mappers and this would allow for a lot of interesting contraptions, while having player-crafted trapdoors be detectable provides some counterplay
Picture

trapdoorlookdemo
Changelog

cl
add: You can now make trapdoors. Craft a trapdoor kit and use it on an openspace tile to make one, then link and activate it with some trapdoor electronics (printable at an autolathe or the engineering lathe) and optionally a trapdoor remote (crafted in personal crafting menu).
qol: You can now unlink trapdoors by using a multitool on them.
balance: Trapdoors now won't break if you just crowbar them. You need to block them from closing, such as with a lattice, cover it up with a wall, or fully destroy the floor tile its on to get rid of them.
/cl

tgstation-server committed:
Automatic changelog generation for PR #68920 [ci skip]

Kapu1178 committed:
Small organ code clean up (#69123)

* Cleans up organ code, removing the EXTERNAL_ORGAN flag, as it can just simply use the external organ subtype instead

Co-authored-by: GoldenAlpharex <58045821+GoldenAlpharex@users.noreply.github.com>

lnGoror committed:
Fixes taste messages (#69035)

* Fixes tastes

* Fixes tastes better

tgstation-server committed:
Automatic changelog generation for PR #69035 [ci skip]

ReinaCoder committed:
pirate captain costume resprite (#69135)

The pirate captain hat and suit has been resprited

tgstation-server committed:
Automatic changelog generation for PR #69135 [ci skip]

CursedBirb committed:
Kobun the breaddog (#68906)

Adds third kitchen craftable animal - Kobun the bread dog. Currently we have Keeki the cak and Terrygold the butter bear. Keeki is best friend of sec due to frosting and Terrygold seems like second clown with its slips, Kobun is designed as help to Psychologist due to powerfull mood aura. Mood granted by aura stays as long as you are within range of aura and when you step out is expires in 10 seconds. You can trade mood aura for a possibility of wearing Kobun as hat

We already have two crafting animals, one made out of cake and another out of butter so why not add another one made out bread? Kobun was made to help people with low mood by giving it mood aura that will bring them back onto their feets

tgstation-server committed:
Automatic changelog generation for PR #68906 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

ShizCalev committed:
Makes glass tables actually transparent (#69189)

They're almost completely opaque right now, to the point where you really can't even see the floor underneath them.

Changed the opacity and redid the legs so you can actually see them

tgstation-server committed:
Automatic changelog generation for PR #69189 [ci skip]

Comxy committed:
New Swap Spell - Wizard (#68895)

* new_swap_spell

* swap copy fixes

* small fixes

* do teleport fix

* secondary target plus fixes

* small sound fix

* More fixes

tgstation-server committed:
Automatic changelog generation for PR #68895 [ci skip]

Time-Green committed:
Geysers no longer need plungers (#69185)

tgstation-server committed:
Automatic changelog generation for PR #69185 [ci skip]

ShizCalev committed:
mood runtime fix (#69197)

FernandoJ8 committed:
De-hardcodes randomize_human() and fixes some related issues along the way (#68876)

* First draft on this branch. Should work.

* Whoopsie

* Some fixes

* And again

* Final draft, question mark?

* Please enter the commit message for your changes

* Uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

* please work

* Saving for the week offline

* Final draft

* Final final draft

* Oh and clean this up

* eah

* Okay, final for real.

* I lied.

* Sure thing boss

* clinclin

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

* That's all of em I think

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68876 [ci skip]

tralezab committed:
Simple flying element typecast thing (#69191)

* Simple flying now doesn't check for simplemob.

ShizCalev committed:
Mecha spark system runtime on destruction. Fixes mecha draining infinite powercells (#69038)

* Mecha spark system runtime on destruction. Fixes mecha draining infinite powercells

tgstation-server committed:
Automatic changelog generation for PR #69038 [ci skip]

JohnFulpWillard committed:
fixes master (#69220)

ArcaneMusic committed:
Cigarette Sprite Update by Kokonut (#69195)

Updates Cigarette boxes, matches (lit and burnt states), and cigar case sprites to 3/4ths perspective

tgstation-server committed:
Automatic changelog generation for PR #69195 [ci skip]

necromanceranne committed:
Reverts #49103 Krav Maga: Stompies Edition (mostly) (#69204)

Reverts Krav Maga to a state prior to #49103 with only some minor retention. That is, leg sweep is still a knockdown.

tgstation-server committed:
Automatic changelog generation for PR #69204 [ci skip]

JohnFulpWillard committed:
Fixes clownborg not fixing their flash (#69203)

Clownborg now calls parent on respawn_consumable, so they will now actually repair their flash like all other cyborgs. I additionally added SHOULD_CALL_PARENT to respawn_consumable to prevent this from happening again in the future.

tgstation-server committed:
Automatic changelog generation for PR #69203 [ci skip]

RandomGamer123 committed:
Fix RCD deconstruction message for turfs being incorrect (#69090)

* Fix RCD deconstruction message for turfs being incorrect

* Update code/game/turfs/open/floor.dm

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69090 [ci skip]

ShizCalev committed:
Permit to perform hazardous work typo fix (#69202)

AnturK committed:
Disables browserlist update message (#69209)

We only care about IE functionality here, the data updates are pretty irrelevant for us and it's confusing for new coders.

ShizCalev committed:
Fixes runtime when attacking liquid plasma with anything that isn't a cup.... (#69199)

* Fixes runtime when attacking liquid plasma with anything that isn't a cup....

GoldenAlpharex committed:
Fixes chat log saving breaking because of unsupported CSS rules (#69211)

tgstation-server committed:
Automatic changelog generation for PR #69211 [ci skip]

ShizCalev committed:
Fixes runtime when ghosts examine artwork (#69200)

Fixed a runtime caused by ghosts examining artwork. Your ancestors will no longer be able to admire your crayon drawings, sorry.

tgstation-server committed:
Automatic changelog generation for PR #69200 [ci skip]

Wallemations committed:
Updates the newscaster wallframe sprite (#69096)

imageadd: Updates the newscaster wallframe sprite

tgstation-server committed:
Automatic changelog generation for PR #69096 [ci skip]

Wallemations committed:
Updates the book scanner sprite (#69176)

Improves the book scanner sprite and adds a sound when scanning a book

tgstation-server committed:
Automatic changelog generation for PR #69176 [ci skip]

CapybaraExtravagante committed:
Ambience buzzing no longer refreshes when you move into areas with the same buzzing sound (#69002)

* Fixes refreshing buzzes

* Forgot to remove some lines

* Update areas.dm

Co-authored-by: Capybara <Capybara@CapybaraMailingServices.com>

tgstation-server committed:
Automatic changelog generation for PR #69002 [ci skip]

MrMelbert committed:
Moves the fax machine in Meta-Med (#69215)

Moves the fax machine in meta-med

tgstation-server committed:
Automatic changelog generation for PR #69215 [ci skip]

timothymtorres committed:
Add liquid dark matter to bluespace bananas (#69207)

* Add liquid dark matter to bluespace bananas

* Buff liquid dark matter

* Add antimatter to taste description

tgstation-server committed:
Automatic changelog generation for PR #69207 [ci skip]

dragomagol committed:
Log lavaland elite creation (#69206)

log lavaland elite creation

Co-authored-by: tattle <article.disaster@gmail.com>

tgstation-server committed:
Automatic changelog generation for PR #69206 [ci skip]

ShizCalev committed:
Fixes "My human left arm itches" message (#69213)

* Fixes "My Human Left arm itches"

* lets get this too

Spookuni committed:
Replaces non-departmental required experisci experiments with extremely expensive discount experiments (#69205)

* Required to expensive

* Historical comment removal

* rolls back departmental experiments to required

* fixes costs for rollback

tgstation-server committed:
Automatic changelog generation for PR #69205 [ci skip]

Maurukas committed:
Fix unsanitary conditions on the Cere emergency shuttle (#69102)

Swaps two tiles so that the cryo gas can be wrenched down
Replaces regular wrench with medical wrench

tgstation-server committed:
Automatic changelog generation for PR #69102 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

EOBGames committed:
The Ambrosia of the Corporate Masses: Coffeemakers (#68178)

Adds coffeemakers to the game. These brew large quantities of coffee at a time, dispense cups and accoutrements, and work from cartridges.

tgstation-server committed:
Automatic changelog generation for PR #68178 [ci skip]

Rhials committed:
Fixes messages for PAI hacking cable extension (#69217)

Updates the visible messages for the PAI's extend_cable proc. They no longer speak to everyone who can see it in the first person.

tgstation-server committed:
Automatic changelog generation for PR #69217 [ci skip]

necromanceranne committed:
Stops rats spawning two every time they die. (#69223)

tgstation-server committed:
Automatic changelog generation for PR #69223 [ci skip]

ShizCalev committed:
Fixes runtime when someone removes a printer's toner cartridge while it's printing (#69198)

tgstation-server committed:
Automatic changelog generation for PR #69198 [ci skip]

RandomGamer123 committed:
Kudzu Fixes v4, hopefully the last one: Fix kudzu possibly obtaining duplicate mutations when being planted from a seed that already has mutations (#68821)

See title, parent mutations that come from vines that came from seeds which came from a different master controller, hence the different addresses.

tgstation-server committed:
Automatic changelog generation for PR #68821 [ci skip]

MrMelbert committed:
Lowers the in-repo threat cost of heretics + also their requirements (#69216)

* lowers the threat cost / pop of heretics

* Adjusts heretic smuggler too

tgstation-server committed:
Automatic changelog generation for PR #69216 [ci skip]

Kapu1178 committed:
Cleans up update_icons, makes the update_icon_updates_onmob element bespoke, updates CODEOWNERS (#69179)

* I just realised this is all one commit.

* hail marry

* fix.

* FIXES IT FOR REAL

* Update code/datums/elements/update_icon_updates_onmob.dm

Pickle-Coding committed:
Significant rework to hydrogen/tritium combustion. (#68230)

Changes hydrogen and tritium combustion gas reactions. Hydrogen/tritium will get consumed at 1 times the burn rate, while oxygen gets consumed at 0.5 times the burn rate, to produce 1 mol of water vapour per burn rate. Significantly changes how the reaction efficiency works. Instead of being an if check on whether there is more oxygen than hydrogen/tritium, then switching between high energy or low energy burn modes, the burn efficiency will instead just be the minimum of half the hydrogen/tritium and 1/20th of the oxygen. The reaction will burn incredibly fast and can end quickly if there is enough oxygen. The energy release of the reactions have been reduced by 90%, which makes them slightly less effective than plasma in total energy gain, but due to their reaction speed, can still be more dangerous and effective for bombs.

As a consequence for the reaction speed, you will collect less tritium when making it as more of the tritium gets consumed before getting scrubbed. The optimal bomb mix has also been changed due to the reaction changes. Generally around 12.85% tritium/hydrogen in a tritium/hydrogen + oxygen mix will give you the best results, filling the tritium or hydrogen to 325.5kPa and oxygen max pressure with portable pump. This does not account of the hot plasma you put in which might change it slightly.

Fixing the stoichiometry of hydrogen and tritium combustion means there won't be weird gas duplication and deletion when you combust then electrolyse them, and will make power balance easier. The weird binary high energy low energy burn modes are a relic of the ancient past, are confusing to understand and is just generally overcomplicated, and doesn't even achieve its job of preventing singlecaps (it actually allows classic singlecaps to be timed). The new combustion formula is much more intuitive to understand, and combined with the lower energy release, changes its purpose in combustion to be a fast reaction that releases energy quickly, rather than just being a better plasma with a weird if check.

Changes hydrogen and tritium gas reaction stoichiometry. 1 mol of hydrogen/tritium and 0.5 mols of oxygen getting consumed will produce 1 mol of water vapour.
Hydrogen and tritium burn rate has been changed to be the minimum of half the hydrogen/tritium, or 1/20th of the oxygen, rather than a binary if check on whether there is more oxygen or not.
Hydrogen and tritium energy release has been reduced by 90%.

tgstation-server committed:
Automatic changelog generation for PR #68230 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

kreindo committed:
Adding tempeh (tem · pay) (#68359)

* Added tempeh

* added tempeh block, tempeh block is processable to tempeh slices, added processor recipe

* added tempeh sprite

* add missing =

* Added tempeh blend for making tempeh

* fixed a variable

* added a temp food_mixture recipe for tempehblend

* fixed a reference

* commenting something

* fixing tempeh slices

* added dots

* added sprite for all the other instances of tempeh

* added 'proper' recipe and revises description

* added actual recipe. (wip)

* finalized recipe, modified icon state name

* icon update

* merged icon

* fixing icon

* resolve conflict... i hope

* removed a file

tgstation-server committed:
Automatic changelog generation for PR #68359 [ci skip]

TerraGS committed:
Fix for gutlunch not eating (#69242)

* fixes gutlunch not going after target objects

* Update code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #69242 [ci skip]

MrMelbert committed:
Fixes Social Anxiety capitalization mid-sentence words (#69233)

Fixes social anxiety having auto-capitalization

tgstation-server committed:
Automatic changelog generation for PR #69233 [ci skip]

dragomagol committed:
Makes holopads display the proper sprite when open (#69232)

* makes holopads display the proper sprite when open

* tabs

Co-authored-by: tattle <article.disaster@gmail.com>

tgstation-server committed:
Automatic changelog generation for PR #69232 [ci skip]

Kapu1178 committed:
It's 2 am and im having a manic episode so i fixed hair (#69092)

* 2 am coding

* lazylists like this are stupid

* reviews

tgstation-server committed:
Automatic changelog generation for PR #69092 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

JohnFulpWillard committed:
Fixes Hilbert Hotel's 'note to the institute' and Cybersun's password (#69237)

* Fixes Hilbert Hotel's paper

Initialize has to set the raw text before calling parent, as paper's base Initialize is when it checks for raw text to add onto the paper.

All other papers already follow this standard, this was just the odd one out.

* I lied, there was another case.

tgstation-server committed:
Automatic changelog generation for PR #69237 [ci skip]

Commit:
Nitrium no longer grants stun immunity (#69180)

Nitrium no longer grants TRAIT_STUNIMMUNE

tgstation-server committed:
Automatic changelog generation for PR #69180 [ci skip]

CursedBirb committed:
[GBP No Update] Smol brain forget that bananas have side sprite (#69101)

* It does not want to let me to do it otherwise

* I hope it works now

tgstation-server committed:
Automatic changelog generation for PR #69101 [ci skip]

tralezab committed:
Necropolis Tendrils now need to be interacted with to drop their loot, and multiple people can receive multiple chests. (#69193)

Tendrils no longer immediately drop their chests, they will drop it once you have used an empty hand on the collapsing tendril. You can only spawn a chest once, but every unique mob that interacts can spawn one chest.

tgstation-server committed:
Automatic changelog generation for PR #69193 [ci skip]

tralezab committed:
AI improvements regarding taming and riding! (#69247)

* Cow AI improvevments and fixes for tamability and riding, no longer allowing you to grab the cow you're riding, and stopping AI cows from moving while being ridden.

tgstation-server committed:
Automatic changelog generation for PR #69247 [ci skip]

Jacquerel committed:
Pun Pun will not save the name "Unknown" to legacy data (#69274)

About The Pull Request

Fixes #69266
If Pun Pun were gibbed in a previous round and ended the next round either wearing a mask or husked, it would save their display name "unknown" to the ancestor data file rather than their real name.

tgstation-server committed:
Automatic changelog generation for PR #69274 [ci skip]

Rhials committed:
Tiziran Canned Goods Pack now supplies the proper supply pack (#69270)

can

c

tgstation-server committed:
Automatic changelog generation for PR #69270 [ci skip]

tralezab committed:
Fixes a fair few execute() errors in dynamic midround rulesets (#69130)

tgstation-server committed:
Automatic changelog generation for PR #69130 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

JohnFulpWillard commented:
I am now interested in this PR

JohnFulpWillard commented on code/modules/reagents/chemistry/reagents/atmos_gas_reagents.dm:
```diff
@@ -0,0 +1,160 @@
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED|REAGENT_NO_RANDOM_RECIPE
+	addiction_types = list(/datum/addiction/stimulants = 14)
+
+/datum/reagent/nitrium_high_metabolization/on_mob_metabolize(mob/living/L)
```
Since you're messing with these, can you remove single letter vars

JohnFulpWillard commented on code/modules/reagents/chemistry/reagents/atmos_gas_reagents.dm:
```diff
@@ -0,0 +1,160 @@
+
+/datum/reagent/pluoxium/on_mob_life(mob/living/carbon/breather, delta_time, times_fired)
+	if(!HAS_TRAIT(breather, TRAIT_KNOCKEDOUT))
+		..()
+		return
```
Can you change this to ``return ..()`` instead

JohnFulpWillard commented on code/modules/reagents/chemistry/reagents/atmos_gas_reagents.dm:
```diff
@@ -0,0 +1,160 @@
+		..()
+		return
+
+	for(var/organ in breather.internal_organs)
+		var/obj/item/organ/organ_being_healed = organ
```
make use of ``as anything`` here instead, please.

GuillaumePrata committed:
Merge remote-tracking branch 'origin/pluoxium_organ_healing' into pluox-stuff-v2

GuillaumePrata committed:
Kills single letter var and a few other things.

GuillaumePrata commented:
Hahaha, I love fucking up github!
I will close this and open a new PR, sorry for the mess.
